app/models/student.rb

```diff
@@ -8,4 +8,12 @@ class Student < ApplicationRecord
   has_many :enrollments, dependent: :destroy
   has_many :grades, through: :enrollments
   has_many :courses, through: :enrollments
+
+  scope :index_from_year, -> (year) {
+    joins("LEFT JOIN (SELECT e.code, c.name, c.year, e.student_id
+      FROM enrollments e
+     JOIN courses c ON c.id = e.course_id AND c.year = #{year}) AS active_enrollments ON active_enrollments.student_id = students.id")
+    .order("students.name, active_enrollments.year")
+    .select("students.id AS id, active_enrollments.code AS enrollment_code, students.name AS student_name, active_enrollments.name AS course_name, active_enrollments.year")
+  }
```

Comment:
Same for this one. Also, I think it would be a good idea to explicitly define the order as `ASC` or `DESC`.

Reply:
I've tried to break this scope into three others because they could be used individually. Do you still think that in this case, it would be a good idea to leave it as a scope? Or would it be better to turn it into a class method?

```ruby
  scope :with_enrollment_details_from_year, lambda { |year|
    joins("LEFT JOIN (#{Enrollment.by_course.by_year(year).to_sql}) temp on temp.student_id = students.id")
      .order('students.name ASC')
      .select('students.id AS id, temp.enrollment_code AS enrollment_code, students.name AS student_name, temp.course AS course_name')
  }
```